packages/aws-cdk-lib/aws-kms/README.md

```diff
@@ -123,6 +123,22 @@ Note that a call to `.addToResourcePolicy(statement)` on `myKeyLookup` will not
 an affect on the key's policy because it is not owned by your stack. The call
 will be a no-op.
 
+If the target key is not found in your account, an error will be thrown.
+To prevent the error in the case, you can receive a dummy key without the error
+by setting `returnDummyKeyOnMissing` to `true`. The dummy key has a `keyId` of
+`1234abcd-12ab-34cd-56ef-1234567890ab`.
+
+```ts
+const dummy = kms.Key.fromLookup(this, 'MyKeyLookup', {
+  aliasName: 'alias/NonExistentAlias',
+  returnDummyKeyOnMissing: true,
+});
+
+if (dummy.keyId === '1234abcd-12ab-34cd-56ef-1234567890ab') {
```

Comment:
Is there a more user-friendly way to allow detecting dummy results? Maybe the `1234abcd-12ab-34cd-56ef-1234567890ab` value can be exposed as a constant from the KMS package (something like `kms.Key.DUMMY_KEY_ID`)? Or potentially even more friendly would be a method like:

```ts
public get isLookupDummy() {
  return this.keyId === Key.DEFAULT_DUMMY_KEY_ID;
}
```

And then in the future if the implementation were to change to, for example, allow more dynamic dummy values (such as what the SSM parameter does) that could be done without a breaking API change.

Reply:
That's a good idea! I will add a new static variable `DEFAULT_DUMMY_KEY_ID` to the `Key` class.

However, as this dummy id is used by an imported key (`IKey`), the `isLookupDummy` method needs to be added to the interface as well as the `KeyBase`. But it would be a breaking change. This is because users who implement their own constructs using the interface will get an error.

Reply:
The following approach is possible, but is this necessary enough to make it a static method?

```ts
export class Key extends KeyBase {
  // ...
  // ...
  public static isLookupDummy(key: IKey): boolean {
    return key.keyId === Key.DEFAULT_DUMMY_KEY_ID;
  }
```

I didn't add this for now, just added DEFAULT_DUMMY_KEY_ID:

https://github.com/aws/aws-cdk/pull/31676/commits/9e8a164f071608faa1f832dabe9213d5cc65cbe1

Reply:
Oh good call on the interface issue!

I do in a way like the static method. It's almost reminiscent of, for example, `iam.Role.isRole`. I could envision that in the future (hypothetically, it'd be possible to someday detect this instead in `fromLookup` and return a class other than `Import`) the body would instead read something like:

```ts
public static isLookupDummy(key: IKey): boolean {
  return LOOKUP_DUMMY_SYMBOL in key;
}
```

In this case, the method actually allows us to make that change (otherwise, we'd be locked to the documented `keyId`-based approach).

This non-failing dummy result for context lookups just feels a bit new overall and I think unlike the SSM parameter situation, detecting this case is actually pretty impactful. We may want to give ourselves some flexibility via abstractions.

But then again, I see and appreciate how simple and straightforward `DEFAULT_DUMMY_KEY_ID` is; so I am not going to withhold an approval over this. If you feel that the static method may actually be _too much_ abstraction for now, I'll happily go ahead and add my approval 😄

Reply:
> In this case, the method actually allows us to make that change (otherwise, we'd be locked to the documented keyId-based approach).

Yeah, agree with you.

I have changed, could you please take a look at it?

Reply:
Thanks! I like that! I appreciate you making the adjustments.